core/libp2p/crypto/key_generator/key_generator_impl.cpp

```diff
@@ -38,6 +38,7 @@ namespace libp2p::crypto {
     outcome::result<std::vector<uint8_t>> encodeKeyDer(KeyStructure *ks,
                                                        Function *function) {
       gsl::owner<unsigned char *> buffer = nullptr;
+      // NOLINTNEXTLINE(cppcoreguidelines-owning-memory)
       auto cleanup = gsl::finally([buffer]() { free(buffer); });
```

Comment:
maybe use `delete` instead of `free`, and remove NOLINTNEXTLINE?

Reply:
or even, use std::unique_ptr with custom deleter?

Reply:
buffer is created somewhere inside of openssl, most probably allocated with malloc();
[they say](https://isocpp.org/wiki/faq/freestore-mgmt#mixing-malloc-and-delete) it would be totally wrong.